src/classes/RD2_ElevateIntegrationService.cls

```diff
@@ -189,4 +203,32 @@ public inherited sharing class RD2_ElevateIntegrationService {
             }
         }
     }
+
+    /**
+    * @description Send Http Request to Elevate and close Closed RD's attached Commitments
+    * @param trackedClosedRds Closed RDs with original Close Reason that will be process
+    */
+    public void closeCommitmentsInElevate(List<npe03__Recurring_Donation__c> trackedClosedRds) {
+
+        List<PS_ElevateAPI.ResponseWrapper> responses = new List<PS_ElevateAPI.ResponseWrapper>();
+
+        for (npe03__Recurring_Donation__c elevateClosedRd : trackedClosedRds) {
+            HttpRequest request = new PS_ElevateAPI.RequestBuilder()
+                .withCommitmentId(elevateClosedRd.CommitmentId__c)
+                .withEndpoint(PS_ElevateAPI.ElevateEndpoint.COMMITMENT_CANCEL)
+                .withMethod(PS_ElevateAPI.HttpRequestMethod.POST)
+                .build();
+            
+            responses.add(apiService.sendRequest(request));
+        }
+
+        for (Integer i = 0; i < responses.size(); i ++) {
+            if (responses[i].statusCode != 200) {
```

Comment:
Probably better if response status codes are stored in a constant somewhere.

Reply:
@andrewyu-salesforce , @force2b : this comment will be resolved by using the new `UTIL_Http` class. See PR #5852 .